### PR TITLE
DOC-4452

### PR DESCRIPTION
**Pull request by jamesnocentini**
https://issues.couchbase.com/browse/DOC-4452

### CODE REVIEW

File: modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift [hunk 1]
```diff
@@ -142,7 +142,32 @@ class SampleCodeTest {
...
+        // tag::document-expiration[]
+        // Purge the document one day from now
+        let ttl = Calendar.current.date(byAdding: .day, value: 1, to: Date())
+        try database.setDocumentExpiration(withID: "doc123", expiration: ttl)
+        
+        // Reset expiration
+        try db.setDocumentExpiration(withID: "doc1", expiration: nil)
+        
+        // Query documents that will be expired in less than five minutes
+        let oneDayFromNow = Date(timeIntervalSinceNow: 60 * 5).timeIntervalSince1970
```
Comment (jayahariv) [inline modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift]: probably you mean, `fiveMintsFromNow`?

File: modules/ROOT/examples/csharp/Program.cs [hunk 2]
```diff
@@ -185,6 +185,27 @@ private static void LoadPrebuilt()
...
+            // tag::document-expiration[]
+            // Purge the document one day from now
+            var ttl = DateTimeOffset.UtcNow.AddDays(1);
+            db.SetDocumentExpiration("doc123", ttl);
+
+            // Reset expiration
+            db.SetDocumentExpiration("doc1", null);
+
+            // Query documents that will be expired in less than five minutes
+            var oneDayFromNow = DateTimeOffset.UtcNow.AddMinutes(5).ToUnixTimeMilliseconds();
```
Comment (jayahariv) [inline modules/ROOT/examples/csharp/Program.cs]: same??

File: modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift [hunk 3]
```diff
@@ -142,7 +142,32 @@ class SampleCodeTest {
...
+        let query = QueryBuilder
+            .select(SelectResult.expression(Meta.id))
+            .from(DataSource.database(db))
+            .where(
+                Meta.expiration.lessThan(
+                    Expression.double(oneDayFromNow)
```
Comment (jamesnocentini) [inline modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift]:
```suggestion
                    Expression.double(fiveMinutesFromNow)
```

File: modules/ROOT/examples/objc/code-snippets/SampleCodeTest.m [hunk 4]
```diff
@@ -130,6 +130,31 @@ - (void) dontTestBatchOperations {
...
+    
+    // Reset expiration
+    [database setDocumentExpirationWithID:@"doc1" expiration:nil error: &error];
+    
+    // Query documents that will be expired in less than five minutes
+    NSTimeInterval fiveMintsFromNow = [[NSDate dateWithTimeIntervalSinceNow:60 * 5] timeIntervalSince1970];
+    CBLQuery* query = [CBLQueryBuilder select: @[[CBLQuerySelectResult expression: [CBLQueryMeta id]]]
+                                         from: [CBLQueryDataSource database: database]
+                                        where: [[CBLQueryMeta expiration]
+                                                lessThan: [CBLQueryExpression double: fiveMintsFromNow]]];
```
Comment (jamesnocentini) [inline modules/ROOT/examples/objc/code-snippets/SampleCodeTest.m]:
```suggestion
                                                lessThan: [CBLQueryExpression double: fiveMinutesFromNow]]];
```

File: modules/ROOT/examples/objc/code-snippets/SampleCodeTest.m [hunk 5]
```diff
@@ -130,6 +130,31 @@ - (void) dontTestBatchOperations {
...
+                                                           value: 1
+                                                          toDate: [NSDate date]
+                                                         options: 0];
+    [database setDocumentExpirationWithID:@"doc123" expiration:ttl error:&error];
+    
+    // Reset expiration
+    [database setDocumentExpirationWithID:@"doc1" expiration:nil error: &error];
+    
+    // Query documents that will be expired in less than five minutes
+    NSTimeInterval fiveMintsFromNow = [[NSDate dateWithTimeIntervalSinceNow:60 * 5] timeIntervalSince1970];
```
Comment (jamesnocentini) [inline modules/ROOT/examples/objc/code-snippets/SampleCodeTest.m]:
```suggestion
    NSTimeInterval fiveMinutesFromNow = [[NSDate dateWithTimeIntervalSinceNow:60 * 5] timeIntervalSince1970];
```

File: modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift (hunk 1 above)
Comment (jamesnocentini) [reply modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift]:
```suggestion
        let fiveMinutesFromNow = Date(timeIntervalSinceNow: 60 * 5).timeIntervalSince1970
```

File: modules/ROOT/examples/csharp/Program.cs (hunk 2 above)
Comment (jamesnocentini) [reply modules/ROOT/examples/csharp/Program.cs]:
```suggestion
            var fiveMinutesFromNow = DateTimeOffset.UtcNow.AddMinutes(5).ToUnixTimeMilliseconds();
```

File: modules/ROOT/examples/csharp/Program.cs [hunk 6]
```diff
@@ -185,6 +185,27 @@ private static void LoadPrebuilt()
...
+            var query = QueryBuilder
+                .Select(SelectResult.Expression(Meta.ID))
+                .From(DataSource.Database(db))
+                .Where(Meta.Expiration.LessThan(Expression.Double(oneDayFromNow)));
```
Comment (jamesnocentini) [inline modules/ROOT/examples/csharp/Program.cs]:
```suggestion
                .Where(Meta.Expiration.LessThan(Expression.Double(fiveMinutesFromNow)));
```

**Review by jamesnocentini — COMMENTED** [review]
rename method to `fiveMinutesFromNow`.

File: modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift (hunk 1 above)
Comment (jamesnocentini) [reply modules/ROOT/examples/swift/code-snippets/SampleCodeTest.swift]: @jayahariv Good catch, I've fixed that for all languages.

File: modules/ROOT/pages/csharp.adoc [hunk 7]
```diff
@@ -275,6 +275,10 @@ include::{snippet}[tag=batch,indent=0]
 At the *local* level this operation is still transactional: no other `Database` instances, including ones managed by the replicator can make changes during the execution of the block, and other instances will not see partial changes.
 But Couchbase Mobile is a distributed system, and due to the way replication works, there's no guarantee that Sync Gateway or other devices will receive your changes all at once.
 
+=== Document Expiration
+
+include::{partialsdir}/document-expiration.adoc[]
```
Comment (eric-schneider) [inline modules/ROOT/pages/csharp.adoc:280]: Interesting. I've never included a partial this way. I've always used `include::partial$document-expiration.adoc[]`

https://docs.antora.org/antora/2.0/asciidoc/include-partial/#include-a-partial-from-the-same-module

**Review by eric-schneider — APPROVED** [review]
lgtm

File: modules/ROOT/pages/csharp.adoc (hunk 7 above)
Comment (jamesnocentini) [reply modules/ROOT/pages/csharp.adoc:280]: Yes, I recently found out about `partial$`. Filed https://issues.couchbase.com/browse/DOC-5125 to be consistent with other components.

**Pull request merged**